### DISCUSSION TITLE
Cinnamon Nemo - change / select mimetype icon for all filetypes

### DISCUSSION THREAD

**Discussion by freckhard** [linuxmint/discussions#58] [Ideas]
Hey there,
I have come across this question now about three times on reddit and the [mint forums](https://forums.linuxmint.com/viewtopic.php?t=187927) and there is a (manual) solution, but I don't think it's very intuitive.

In nemo, you have the option to rightclick-properties a file and change its icon. But this then sets it just for this/these particular file(s). If I want to change the icon for - lets say all pdfs - I can't "set the new icon as default for this filetype".

So the idea would be to be able to change the icon on a file and in the icon selection window to offer a checkbox "set default for this filetype".

Thanks in advance!

**Comment by Secret-chest** [comment]
Edit the iconpack, it's not that hard. You will need to swap some filenames, no need to be an artist.

It's not possible to implement this since if we do add it, only Nemo will support it. Not the GTK filepicker or any apps showing directories.

**Reply by freckhard** [reply]
Thank you for your reply. As far as I can tell it is not per se nemo specific, but related to the cinnamon theme. So if not implemented into nemo, it may go in the system settings? (This is just a suggestion, the developers will know the best place where to put that.)
I hope this clarifies what I meant, or did I misunderstand you? Thanks!

**Reply by Secret-chest** [reply]
The devs would consider this overkill. The only way to change the default file icons is to modify the icon theme.

**Reply by freckhard** [reply]
Ah I see our misunderstanding, no I don't want to create a new image and use that, I want to select and map an existing one, i.e. making all pdf files show with the text icon (instead of the pdf icon obviously). The icons are present within the theme and already selectable for a single file. All that's left is to set it for every file of that mime-type.

I can do so easily by symlinking to the desired mime-type, as descried in the above mentioned forum post. That is why it is not overkill, but a small but neat feature I am stating here for discussion :-)

**Reply by Secret-chest** [reply]
It is overkill for the 95% of users.

**Reply by freckhard** [reply]
This feels like a very narrow minded discussion now, maybe I still misunderstand you. In case I don't I think talking on behalf of 95% is presumptuous on your end. To whatever metric is used, I am amongst those 5% then and would like this feature. Hence I opened this discussion. To discuss and not to get shut down with every reply. Or are you speaking on behalf of Clement & Team?

**Back to topic**, I can provide a step by step guide, bash script / pseudo code to achieve the desired outcome for the devs to look at, if desired. Looking forward for constructive feedback! Thanks!

**Reply by JosephMcc** [reply]
> Or are you speaking on behalf of Clement & Team?

No, this person certainly is not and probably shouldn't be putting works in people's mouths.

The point made about every file chooser outside of Nemo is true. Not sure if it's possible but have you considered implementing this as a Nemo action? So you could just right click on a file and change the icon for the type of file you clicked on? If you already know how to do it, it may be possible. There has been some talk in the past of having a "Nemo actions" category or something like that as part of Cinnamon spices.

**Reply by Secret-chest** [reply]
Sorry @JosephMcc , however I do have some knowledge about Gtk and freedesktop. I was just saying why this probably won't be added.
